underscore.js

```diff
@@ -490,17 +490,21 @@
   };
 
   // Internal implementation of a recursive `flatten` function.
-  var flatten = function(input, shallow, strict) {
-    var output = [], idx = 0;
+  var flatten = function(input, shallow, strict, _output, _idx) {
+    var output = _output ? _output : [], idx = _idx ? _idx : 0;
```

Comment:
Instead of passing `_idx` can you just make `idx = output.length`. Otherwise LGTM